store/sqlstore/thread_store.go

```diff
@@ -133,17 +137,57 @@ func (s *SqlThreadStore) GetThreadsForUser(userId string, opts model.GetUserThre
 	if opts.PageSize == 0 {
 		pageSize = opts.PageSize
 	}
+
+	mentionsQuery, mentionsQueryArgs, _ := s.getQueryBuilder().
+		Select("SUM(ThreadMemberships.UnreadMentions)").
+		From("ThreadMemberships").
+		Join("Posts ON Posts.RootId = ThreadMemberships.PostId").
+		Join("Channels ON Posts.ChannelId = Channels.Id").
+		Where(fetchConditions).
+		Where("Posts.UpdateAt >= ThreadMemberships.LastViewed").
+		GroupBy("ThreadMemberships.UserId, ThreadMemberships.Following, Posts.DeleteAt").ToSql()
```

Comment:
I don't think you need grouping here, you just need to ensure that the join for the channels is a left join, to not duplicate rows in any way.